app/collector.go

```diff
@@ -118,7 +118,9 @@ func (c *collector) Report(_ context.Context) (report.Report, error) {
 	}
 
 	c.clean()
-	return c.merger.Merge(c.reports), nil
+	rpt := c.merger.Merge(c.reports).Upgrade()
```

Comment:
Oh yeah, that one. I noticed that `c.cached` was never set to anything but `nil`. I guess it was a bug…